rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -1093,19 +1093,58 @@ namespace rsx
 				}
 				else
 				{
-					if (out_pitch != in_pitch || out_pitch != out_bpp * out_w)
+					const bool is_overlapping = scale_x > 0 && scale_y > 0 && [&]() -> bool
 					{
-						for (u32 y = 0; y < out_h; ++y)
+						const u32 src_max = src_offset + (in_pitch * (in_h - 1) + (in_bpp * in_w)) - 1;
+						const u32 dst_max = dst_offset + (out_pitch * (out_h - 1) + (out_bpp * out_w)) - 1;
+						return (src_offset >= dst_offset && src_offset <= dst_max) ||
```

Comment:
Shouldn't you also compare dma context? This method is mostly used for cross-dma communication afterall.